wrappers/kratosWrapper.py

```diff
@@ -51,68 +47,127 @@ def __init__(self):
         self.SP = DataContainer()
         self.BC = DataContainer()
 
+        # Extended data containers
+        self.SPE = {}
+
         # Initialization
         self.initialize()
 
-    def __addNodalVariablesToModelpart(self):
-        pass
+    # ABCModelingEngine Implementation
+
+    def add_particles(self, particle_container):
+        message = 'KratosWrapper does not handle particle container'
+        raise NotImplementedError(message)
+
+    def add_mesh(self, src):
```

Comment:
Note, my suggestion of what to be done is also faulty as it wouldn't handle renaming of the mesh (```mesh.name="new_name"```) as the wrapper would be unaware of such changes: